contracts/governance/BaseWeightedMultisig.sol

```diff
@@ -179,19 +182,47 @@ abstract contract BaseWeightedMultisig is IBaseWeightedMultisig {
         revert LowSignaturesWeight();
     }
 
+    // /**
+    //  * @dev This function is used to hash the data with auth related data to compute the message hash that will be signed
+    //  * @dev keccak256(domainSeparator || signersHash || dataHash)
+    //  * @param dataHash The hash of the data
+    //  * @param signersHash The hash of the weighted signers that sign off on the data
+    //  * @return The message hash to be signed
+    //  */
+    // function hashMessage(bytes32 dataHash, bytes32 signersHash) public view returns (bytes32) {
+    //     bytes32 messageHash = keccak256(abi.encodePacked(domainSeparator, signersHash, dataHash));
+    //     return ECDSA.toEthSignedMessageHash(messageHash);
+    // }
+
+    /**
+     * @dev Returns an Ethereum Signed Message, created from `dataHash`, `signersHash`, and `domainSeparator`.
+     * This replicates the behavior of the
+     * https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign[`eth_sign`]
+     * JSON-RPC method.
+     *
+     * See {recover}.
+     *
+     * @param signersHash The hash of the weighted signers that sign off on the data
+     * @param dataHash The hash of the data
+     */
+    function hashMessage(bytes32 signersHash, bytes32 dataHash) public view returns (bytes32) {
+        // 96 is the length of the trailing bytes
+        return keccak256(abi.encodePacked('\x19Ethereum Signed Message:\n96', domainSeparator, signersHash, dataHash));
```

Comment:
Somehow I feel signersHash is redundant here

Reply:
Maybe we should keep the `Ethereum Signed Message:` part in the util? But this optimization also makes sense